modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/VirtualScrollBar.java

```diff
@@ -129,18 +129,34 @@ public final BooleanProperty virtualProperty() {
         if (isVirtual()) {
             adjusting = true;
             double oldValue = flow.getPosition();
+            double newValue = ((getMax() - getMin()) * Utils.clamp(0, pos, 1)) + getMin();
+            /**
+             * JDK-8173321: Click on trough has no effect when cell height > viewport height:
+             * Solution: Scroll one cell further up resp. down if only one cell is shown.
+             */
+            IndexedCell firstVisibleCell = flow.getFirstVisibleCell();
+            IndexedCell lastVisibleCell = flow.getLastVisibleCell();
+            if (firstVisibleCell == lastVisibleCell) {
+                int index = firstVisibleCell.getIndex();
+                if (newValue < oldValue) {
+                    flow.scrollTo(index - 1);
+                } else {
+                    flow.scrollTo(index + 1);
+                }
 
-            double newValue = ((getMax() - getMin()) * Utils.clamp(0, pos, 1))+getMin();
-            if (newValue < oldValue) {
-                IndexedCell cell = flow.getFirstVisibleCell();
-                if (cell == null) return;
-                flow.scrollToBottom(cell);
-            } else if (newValue > oldValue) {
-                IndexedCell cell = flow.getLastVisibleCell();
-                if (cell == null) return;
-                flow.scrollToTop(cell);
+            } else {
+                if (newValue < oldValue) {
+                    IndexedCell cell = firstVisibleCell;
+                    if (cell == null)
+                        return;
```

Comment:
Unless the if statement, including its target, is entirely on one line, you _must_ enclose the target of the `if` with curly braces, like this:

```
                    if (cell == null) {
                        return;
                    }
```

Since it fits on one line, the prior code is also acceptable:

```
                    if (cell == null) return;
```

Reply:
I will change this to the first alternative in my next commit.